### PR TITLE
First draft for preproccessing Morocco legislation

### LINKED ISSUE #35: Develop preprocessing script for legislation from Morocco

### PR DESCRIPTION
**Pull request by Skyrolon**
Issues I was not able to resolve:

1. Article 1: I'm not sure how to turn all the '-' into bullet points in an efficient one. I can do them one by one, but I can't figure out how to do replace all in one paragraph.
2. Article 3: I'm not sure how you want me to handle that or if the formatting is fine as it is.
3. Article 3: There is a footnote ("* - Dahir N° 1.03.59 du 10 Rabiï I 1424 (12 mai [2003], BO N° 5118 du Juin 2003 2") that I'm not sure how to remove. I think some of the symbols might be messing it up.\
4. I ran a spellcheck in word and it only detected one missed accent. My french is basically non-existent at this level so I'm not sure if there are more errors or a better spellchecker I should use.

Closes #35

### CODE REVIEW

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 1]
```diff
@@ -0,0 +1,219 @@
+Chapitre Premier - Dispositions générales 
+
+Section Première - Objectifs et principes généraux 
+
+(1) La présente loi a pour objet d'édicter les règles de base et les principes généraux de la politique nationale dans le domaine de la protection et de la mise en valeur de l'environnement. Ces règles et principes visent à: - protéger l'environnement contre toutes formes de pollution et de dégradation quelle qu'en soit l'origine; - améliorer le cadre et les conditions de vie de l'homme; - définir les orientations de base du cadre législatif, technique et financier concernant la protection et la gestion de l'environnement; - mettre en place un régime spécifique de responsabilité garantissant la réparation des dommages causés à l'environnement et l'indemnisation des victimes.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:5]: You can replace all of the dashes with bullet points with the following command:

```
sed -E ':start s/^(\(1\).*)- /\1• /;t start'
```

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
...bonne gestion de l'environnement font partie de la politique intégrée du développement économique, social et culturel; [•] La protection et la mise en valeur de l'environnement constituent une utilité publique et une responsabilité collective nécessitant la participation, l'information et la détermination des responsabilités; [•] L'instauration d'un équilibre nécessaire entre les exigences du développement national et celles de la protection de l'environnement lors de l'élaboration des plans sectoriels de développement et l'intégration du concept du développement durable lors de l'élaboration et de l'exécution de ces plans; [•] La prise en considération de la protection de l'environnement et de l'équilibre écologique lors de l'élaboration et de l'exécution des plans d'aménagement du territoire; [•] La mise en application effective des principes de " l'usager payeur " et " du pollueur payeur " en ce qui concerne la réalisation et la gestion des projets économiques et sociaux et la prestation de services; [•] Le respect des pactes internationaux en matière d'environnement lors de l'élaboration aussi bien des plans et programmes de développement que de la législation environnementale.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:7]: Please remove the unnecessary spacing around the double quotation marks in `" l'usager payeur "` and `" du pollueur payeur "`.

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
...s résidus résultant d'un processus d'extraction, exploitation, transformation, production, consommation, utilisation, contrôle ou filtration, et d'une manière générale, tous objet et matière abandonnés ou que le détenteur doit éliminer pour ne pas porter atteinte à la santé, à la salubrité publique et à l'environnement. 24 - Déchets dangereux: toutes formes de déchets qui, par leur nature dangereuse, toxique, réactive, explosive, inflammable, biologique ou bactérienne, sont susceptibles de constituer un danger pour l'équilibre écologique tel que fixé par les normes internationales dans ce domaine ou contenu dans des annexes complémentaires qui seront fixées par voie réglementaire. 25 - Produits et facteurs polluants: tout produit solide, liquide ou gazeux, bruit, radiations, chaleur ou vibrations sonores résultant des activités humaines et susceptibles, directement ou indirectement, de polluer l'environnement ou de favoriser sa dégradation. 26 - Pollueur: toute personne physique ou morale causant ou participant à un état de pollution. 27 - Espaces maritimes: ressources naturelles maritimes biologiques et minérales du fond de la mer, des eaux avoisinantes ou en dessous du sol marin.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:11]: Please remove all of the numbers and hyphens that precede each defined phrase.

You can remove `* - Dahir N° 1.03.59 du 10 Rabiï I 1424 (12 mai [2003], BO N° 5118 du Juin 2003 2` with the following command:

```
amend_error_in_article '\* - Dahir.*Etablissements humains' 'Établissements humains'
```

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 2 above)
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:7]: Please remove the unnecessary spacing around the double quotation marks in `" l'usager payeur "` and `" du pollueur payeur "`.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 1 above)
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:5]: You can replace all of the dashes with bullet points with the following command:

```
sed -E ':start;s/^(\(1\).*)- /\1• /;t start'
```

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+
+(32) Les dispositions législatives et réglementaires déterminent les mesures à entreprendre en vue de préserver la qualité de l'air ainsi que les normes de contrôle et de suivi nécessaires.
+
+Section V - Les espaces et les ressources marins, y compris le littoral
+
+(33) En vue de la protection des espaces et des ressources marins sous souveraineté ou juridiction nationale, des dispositions législatives et réglementaires sont prises pour prévenir et mettre fin aux activités susceptibles d'altérer la qualité des eaux et des ressources marines, de porter atteinte à la santé de l'homme ou de nuire à la faune, à la flore, aux intérêts connexes et à l'environnement marin et côtier en général.
+
+(34) Les dispositions législatives et réglementaires fixent: [•] les conditions d'exploration, d'exploitation et de mise en valeur des ressources marines; [•] les mesures nécessaires pour la prévention et la lutte contre la pollution marine, y compris celle résultant des accidents maritimes imprévisibles; les critères nécessaires au classement des aires spécialement protégées.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:93]: Is there a bullet point missing before `les critères nécessaires`?

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+
+(35) Pour la protection, la mise en valeur et la conservation du littoral, des dispositions législatives et réglementaires sont prises pour assurer la gestion intégrée et durable de l'écosystème du littoral et la prévention de toute dégradation de ses ressources.
+
+(36) Les dispositions législatives et réglementaires fixent les mécanismes et les moyens de protection des espaces et ressources marins, notamment:- les modalités d'élaboration des schémas et des plans d'aménagement et d'exploitation du littoral; [•] les critères nécessaires au classement d'une partie du littoral en aires spécialement protégées telles que définies par l'article 38 de la présente loi;- les conditions d'exploitation, de mise en valeur et de développement des ressources du littoral.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:97]: Two hyphens need to be replaced with bullet points in this article.

File: raw_data/preprocessed/africa/morocco_french.txt [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+
+Section VI - Les campagnes et les zones montagneuses
+
+(37) En vue de la protection du monde rural, la conservation et la mise en valeur des écosystèmes dans les campagnes et les zones montagneuses, des dispositions législatives et réglementaires sont prises aux fins d'assurer une gestion intégrée et durable des écosystèmes et de les protéger contre toute dégradation de leurs ressources et de la qualité de l'environnement en général. Les dispositions législatives et réglementaires fixent notamment: - les modalités d'élaboration des schémas et plans d'aménagement et de gestion intégrée des campagnes et des zones montagneuses; - les critères nécessaires au classement des campagnes et des zones montagneuses en aires spécialement protégées telles que définies par l'article 38 de la présente loi; - les conditions d'exploitation, de protection et de mise en valeur des ressources des campagnes et des zones montagneuses.
```
Comment (danpdraper) [inline raw_data/preprocessed/africa/morocco_french.txt:101]: The hyphens need to be replaced with bullet points in this article.

File: scripts/legislation/africa/morocco_french.sh [hunk 7]
```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
```
Comment (danpdraper) [inline scripts/legislation/africa/morocco_french.sh:1]: I'll review this file after you have addressed the comments above.

**Review by danpdraper — CHANGES_REQUESTED** [review]

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 4 above)
Comment (Skyrolon) [reply raw_data/preprocessed/africa/morocco_french.txt:93]: The [legislation](http://extwprlegs1.fao.org/docs/pdf/mor42766.pdf) does not contain a bullet point, but I am inclined to believe this is an oversight on their part. Do you want me to add one in?

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 1 above)
Comment (Skyrolon) [reply raw_data/preprocessed/africa/morocco_french.txt:5]: There seems to be some inconsistency between the bullet point formats used in the scripts. I see that [•] is normally used, but here you use • (there are no brackets). Is there a format we should use consistently?

Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:5]: Sorry, there should be square brackets around the bullet point. Please use the following command:

```
sed -E ':start;s/^(\(1\).*)- /\1[•] /;t start'
```

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 4 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:93]: Yes, please.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 1 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:5]: This doesn't appear to have been fixed in the new pull request.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 2 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:7]: This doesn't appear to have been fixed in the new pull request.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 3 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:11]: This provision still contains all of the numbers and hyphens in the new pull request.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 5 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:97]: This doesn't appear to have been fixed in the new pull request.

File: raw_data/preprocessed/africa/morocco_french.txt (hunk 6 above)
Comment (danpdraper) [reply raw_data/preprocessed/africa/morocco_french.txt:101]: This doesn't appear to have been fixed in the new pull request.

**Pull request merged**